enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,357 @@
+`OPENSHIFT_SHARED_CPUS=<CPU-IDs>` will be present under the container's environment variables.
+
+The mixed-cpu-node-plugin itself is essentially a pod that runs as a DaemonSet on the worker nodes.
+cluster-node-tuning-operator (NTO) will be modified to support the deployment of the DaemonSet and additional
+resources (ServiceAccount, RBAC resources, SecurityContextConstraint, etc.)
+NTO will also be responsible to watch, monitor and report the node-plugin's DaemonSet status.
+Additionally, since the plugin is part of OCP infrastructure pods, NTO annotates the pods with:
+```yaml
+target.workload.openshift.io/management: {
+  "effect":"PreferredDuringScheduling"
+}
+```
+See https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md for details.
+
+Since both the components are related,
+mixed-cpu-node-plugin [source code](https://github.com/openshift-kni/mixed-cpu-node-plugin) would be vendored under NTO.
+By having different main.go + container entrypoint, NTO image can be also used as the
+mixed-cpu-node-plugin image. 
+This way saves the overhead of having a separate container image with all the maintenance cost that comes with it.
+
+In order to activate NRI in container-runtime, NTO creates MachineConfig to enable NRI in 
+the container-runtime configuration:
+```yaml
+[crio.nri]
+enable_nri = true
+```
+
+The plugin is completely optional and is off by default.
+
+No changes are needed in the container-runtime, because NRI hooks already supported by the runtime
+and mixed-cpu-node-plugin implementing the NRI interface.
+This minimizes risks and helps avoid regression on
+a container-runtime component (CRI-O in OpenShift case).
+
+_"A picture is worth a thousand words"_
+![mixed-cpu-node-plugin](mixed-cpu-node-plugin.jpg)
+
+### API
+
+Extend PerformanceProfile with new `workloadHints` named `mixedCpus` in order to enable
+the feature.
+
+```yaml
+workloadHints:
+  mixedCpus: true
+```
+
+Specify the shared cpus via annotation:
+
+```yaml
+annotations:
+    mixed-cpu.openshift.io/shared-cpus: "2-3"
+```
+
+The CPU set value should follow the 
+[cpuset](https://github.com/openshift/cluster-node-tuning-operator/blob/master/docs/performanceprofile/performance_profile.md#cpuset) conventions.
+It defines a set of CPUs that can be allocated to guaranteed pods/containers that still require
+non-exclusive ones.
+
+The CPU set must not overlap with `spec.cpu.reserved` or `spec.cpu.isolated` and NTO
+should return an error in case it does.
+
+The reason for using an annotation and not new API field 
+is because there is a coming feature named shared-partition
+that uses similar convention i.e., shared cpus, and would be 
+enabled via NTO as well.
+
+Once shared-partition lands, 
+the annotation should be [deprecated](mixed-cpu-node-plugin.md#removing-a-deprecated-feature)
+and node-plugin will use the shared cpus specified 
+at `spec.cpu.shared`.
```

Comment:
So why not just wait for this new feature?

Reply:
This feature needed for 4.14 and would probably postpone to 4.15 due to review capacity, while shared-partition aims for 4.15 as well but we need the annotation as safety mechanism in case shared-partition won't make it for 4.15 timeline.
In any case this is a valid point, please check https://github.com/openshift/enhancements/pull/1396#discussion_r1251980304